### ISSUE TITLE
Web builds do not tree-shake icons by default

### ISSUE THREAD

**Issue by guidezpl** [flutter/flutter#102490]
This means web apps (e.g. https://gallery-flutter-staging.web.app/) using material and cupertino icons include:
- `MaterialIcons-Regular.otf`, which is now 1.6MB and 410KB compressed
- `CupertinoIcons.ttf`, which is 283KB and 110KB compressed

Blocked on https://github.com/flutter/flutter/issues/57181

**Comment by guidezpl** [comment]
Closing in favor of #90171
